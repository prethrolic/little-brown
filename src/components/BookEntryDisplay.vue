<template>
  <div
    :class="['bed-container', { 'clicked': clicked }]"
    @click='onClick'
    @animationend='clicked = false'
  >
    <div class='bed-cover-img-container'>
      <img class='bed-cover-img' :src="book.cover">
    </div>
    <div class='bed-detail-container'>
      <div v-html="book.title" class='bed-title' />
      <div v-if="book.isPromoted" class='bed-promo-tag'>promo</div>
      <div v-html="'à¸¿' + book.price" class='bed-price' />
    </div>
  </div>
</template>
<script>
export default {
  name: 'BookEntryDisplay',
  props: {
    book: Object,
  },
  data() {
    return {
      clicked: false,
    };
  },
  methods: {
    onClick() {
      this.$store.commit('add_to_cart', this.book);
      this.clicked = true;
    },
  },
};
</script>
<style lang="scss" scoped>
@import '@/assets/scss/variables.scss';

.bed {
  &-container {
    display: grid;
    align-items: center;
    grid-template-columns: 1fr 2fr;
    grid-auto-rows: 1fr;
    height: 100%;
    padding: 12px;
    border-radius: 8px;
    -webkit-box-shadow: 0px 0px 21px -5px rgba(199,199,199,0.44);
    box-shadow: 0px 0px 21px -5px rgba(199,199,199,0.44);
    transition: 0.2s ease-in-out;
    cursor: pointer;

    &:hover {
      -webkit-box-shadow: 0px 0px 21px -5px rgba(46,46,46,0.44);
      box-shadow: 0px 0px 21px -5px rgba(46,46,46,0.44);
    }

    &.clicked {
      -webkit-box-shadow: 0px 0px 21px -5px rgba(199,199,199,0.44);
      box-shadow: 0px 0px 21px -5px rgba(199,199,199,0.44);
      animation: 0.4s bounce ease-in-out;
    }
  }

  &-cover-img {
    max-width: 100%;
    border-radius: 8px;
  }

  &-promo-tag {
    width: fit-content;
    text-transform: uppercase;
    font-size: 10px;
    font-weight: bold;
    padding: 2px 8px;
    margin-top: 8px;
    border-radius:4px;
    background-color: $primary;
    color: white;
  }

  &-detail-container {
    height: 100%;
    display: flex;
    flex-direction: column;
    text-align: left;
    margin-left: 4px;
    padding: 8px;
  }

  &-title {
    font-size: 12px;
    display: -webkit-box;
    font-weight: bold;
    text-transform: uppercase;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
    word-wrap: break-word;
  }

  &-price {
    margin-top: auto;
  }
}
</style>
