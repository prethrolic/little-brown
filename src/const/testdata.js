export const bookList = [
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855652.jpg",
    "price": "350",
    "title": "Harry Potter and the Philosopher's Stone (I)",
    "id": "9781408855652"
  },
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855669.jpg",
    "price": "350",
    "title": "Harry Potter and the Chamber of Secrets (II)",
    "id": "9781408855669"
  },
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855676.jpg",
    "price": "340",
    "title": "Harry Potter and the Prisoner of Azkaban (III)",
    "id": "9781408855676"
  },
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9780/2419/9780241975367.jpg",
    "price": "230",
    "title": "The Monk of Mokha",
    "id": "9780241975367"
  },
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/7847/9781784706883.jpg",
    "price": "300",
    "title": "The Language of Kindness",
    "id": "9781784706883"
  },
];

export const cart = {
  "9781408855652": {
    book: {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855652.jpg",
    "price": "350",
    "title": "Harry Potter and the Philosopher's Stone (I)",
    "id": "9781408855652"
    },
    count: 2,
  },
  "9781408855669": {
    book: {
      "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855669.jpg",
      "price": "350",
      "title": "Harry Potter and the Chamber of Secrets (II)",
      "id": "9781408855669"
    },
    count: 1,
  },
  "9781408855676": {
    book: {
      "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855676.jpg",
      "price": "340",
      "title": "Harry Potter and the Prisoner of Azkaban (III)",
      "id": "9781408855676"
    },
    count: 1,
  },
  "9780241975367": {
    book: {
      "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9780/2419/9780241975367.jpg",
      "price": "230",
      "title": "The Monk of Mokha",
      "id": "9780241975367"
    },
    count: 2,
  },
};

export const inPromotion = ["9781408855652", "9781408855669", "9781408855676"];

export const user = {
  displayName: 'John Doe',
  pictureUrl: 'https://pbs.twimg.com/profile_images/1269585418107805697/6QKUkoWJ_400x400.jpg',
};

export const paymentInfo = {
  subtotal: '1850.00',
  discount: '114.40',
  total: '1735.60',
  cash: '2000',
};