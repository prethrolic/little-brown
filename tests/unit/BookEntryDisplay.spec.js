import { shallowMount } from '@vue/test-utils';
import BookEntryDisplay from '@/components/BookEntryDisplay';
import { bookList } from '@/const/testdata.js';

describe('BookEntryDisplay.vue', () => {
  it('render book details correctly', () => {
    const wrapper = shallowMount(BookEntryDisplay, {
      propsData: {
        book: bookList[0],
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});