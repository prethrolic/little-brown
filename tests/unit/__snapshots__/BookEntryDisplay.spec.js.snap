// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookEntryDisplay.vue render book details correctly 1`] = `
<div
  class="bed-container"
>
  <div
    class="bed-cover-img-container"
  >
    <img
      class="bed-cover-img"
      src="https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855652.jpg"
    />
  </div>
   
  <div
    class="bed-detail-container"
  >
    <div
      class="bed-title"
    >
      Harry Potter and the Philosopher's Stone (I)
    </div>
     
    <!---->
     
    <div
      class="bed-price"
    >
      à¸¿350
    </div>
  </div>
</div>
`;
